src/types.ts

```diff
@@ -142,11 +142,37 @@ export type TransactionRequest = {
     chainId?: unknown,
 }
 
+export interface EventData {
+  address: string,
+  blockHash: string,
+  blockNumber: string,
+  data: string,
+  logIndex: string,
+  removed: boolean,
+  topics: string[],
+  transactionHash: string,
+  transactionIndex: string
+};
+
+
+export interface EventFilter {
+  address?: string;
+  topics?: Array<string | Array<string>>;
+}
+
+export type BlockTag = string | number;
```

Comment:
i don't think that `fromBlock` or `toBlock` can be passed to `eth_getLogs` as a number